### PR TITLE
CCS paper

### PR DESCRIPTION
**Pull request by dionyziz**

### CODE REVIEW

File: etc/ccs/theory.tex [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+\begin{document}
+
+\maketitle
+
+\section*{Security under adaptive reflection}
+
+\subsection*{Adaptive reflection game}
+
+The game $\text{Game}_{\text{IND-REF}}^{\mathcal{A}, f,
+\mathcal{V}}(1^\lambda)$ is parameterized with a rendering function $f(\cdot,
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: It is ugly that we have to parameterize the game with the rendering function f and the noise distribution V. Ideally, we'd want the theorems arising to be true for some classes of f and V, if not for all.

File: etc/ccs/theory.tex [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+adversary is successful if $s = s^*$.
+
+\subsection*{Adaptive reflection security}
+
+Given a rendering function $f(\cdot, \cdot, \cdot)$ and a noise distribution
+$\mathcal{V}$, an encryption function $\mathcal{E}$ is
+\textit{indistinguishable under adaptive reflection} IND-REF if:
+
+$\forall PPT \mathcal{A}: |P[\text{Game}_{\text{IND-REF}}^{\mathcal{A}, f,
+\mathcal{V}}(1^\lambda) = 1] - P_{s_1 \leftarrow \mathcal{S}, s_2 \leftarrow
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: The use of distribution S here is out of scope, as it is chosen by the adversary in a per-game basis. How can we express this better?

File: etc/ccs/theory.tex [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+\subsection*{Adaptive reflection game}
+
+The game $\text{Game}_{\text{IND-REF}}^{\mathcal{A}, f,
+\mathcal{V}}(1^\lambda)$ is parameterized with a rendering function $f(\cdot,
+\cdot, \cdot)$, a noise distribution $\mathcal{V}$, a security parameter
+$\lambda$ and an encryption function $\mathcal{E}$.
+
+At the initialization phase of the game, the challenger produces a random
+symmetric $\lambda$-bit key $\kappa \leftarrow \mathcal{K}$. The adversary
```
Comment (dimkarakostas) [inline etc/ccs/theory.tex]: define K?

File: etc/ccs/theory.tex (hunk 2 above)
Comment (dionyziz) [inline etc/ccs/theory.tex]: To fix out-of-scope problem, move distribution `S` in `\forall` quantifier in front of PPT A quantifier, parameterize game with `S`.

File: etc/ccs/theory.tex [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+\subsection*{Adaptive reflection security}
+
+Given a rendering function $f(\cdot, \cdot, \cdot)$ and a noise distribution
+$\mathcal{V}$, an encryption function $\mathcal{E}$ is
+\textit{indistinguishable under adaptive reflection} IND-REF if:
+
+$\forall PPT \mathcal{A}: |P[\text{Game}_{\text{IND-REF}}^{\mathcal{A}, f,
+\mathcal{V}}(1^\lambda) = 1] - P_{s_1 \leftarrow \mathcal{S}, s_2 \leftarrow
+\mathcal{S}}[s_1 = s_2]| = negl(\lambda)$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: This definition is problematic. If the adversary chooses `S = { α with P = 1, β with P = 0 }` and then tries guess `s* = β`, then the probability of succeeding in the game is 0, but the probability difference is 1, hence the system is not IND-REF secure. Perhaps try using entropy?

File: etc/ccs/theory.tex [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+\end{document}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: +section with compression

File: etc/ccs/theory.tex [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+\subsection*{Adaptive reflection game}
+
+The game $\text{Game}_{\text{IND-REF}}^{\mathcal{A}, f,
+\mathcal{V}}(1^\lambda)$ is parameterized with a security parameter $\lambda$,
+a polynomial in $\lambda$ rendering function $f(\cdot, \cdot, \cdot)$, a noise
+distribution $\mathcal{V}$, and an encryption function $\mathcal{E}$.
+
+At the initialization phase of the game, the challenger produces a random
+symmetric $\lambda$-bit key $\kappa \leftarrow \mathcal{K}$. The adversary is
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: This is modelled as a symmetric encryption, but there's nothing prohibiting us from using a similar model for asymmetric encryption.

File: etc/ccs/theory.tex [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+\maketitle
+
+\section*{Security under adaptive reflection}
+
+\subsection*{Adaptive reflection game}
+
+The game $\text{Game}_{\text{IND-REF}}^{\mathcal{A}, f,
+\mathcal{V}}(1^\lambda)$ is parameterized with a security parameter $\lambda$,
+a polynomial in $\lambda$ rendering function $f(\cdot, \cdot, \cdot)$, a noise
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Avoid parameterizing the game with V, f. Use an advantage against an adversary without the secret.

https://www.iacr.org/archive/pkc2003/25670071/25670071.pdf

File: etc/ccs/theory.tex [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+distribution $\mathcal{V}$, and an encryption function $\mathcal{E}$.
+
+At the initialization phase of the game, the challenger produces a random
+symmetric $\lambda$-bit key $\kappa \leftarrow \mathcal{K}$. The adversary is
+given $\mathcal{V}$ and $f(\cdot, \cdot, \cdot)$.  The adversary chooses a
+distribution of secrets $\mathcal{S}$ such that $\forall s_1, s_2 \in
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Use $\mathcal{M}$ instead of $\mathcal{M}$

File: etc/ccs/theory.tex [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+\begin{document}
+
+\maketitle
+
+\section*{Security under adaptive reflection}
+
+\subsection*{Adaptive reflection game}
+
+The game $\text{Game}_{\text{IND-REF}}^{\mathcal{A}, f,
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Don't call the game indistinguishability game, call it semantically secure.

File: etc/ccs/compression.tex [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+the non uniform distribution X. $y = f(x)$ will follow the distribution $f(X)$.
+$E(|y|) / E(|x|) < 1$ is the metric that defines the function's performance.
+
+\subsection*{Leaking compression function}
+Leaking compression function $C$ is a compression function where
+for each data string x there exist $x_1$, $x_2$ such that:
+
+$|x_1|==|x_2|$ and
+$|C(n_1||x||n_2||x_1||n_3)| < |C(n_1||x||n_2||x_2||n_3)|$,
+where $n_1$, $n_2$, $n_3$ random nonces.
```
Comment (dionyziz) [inline etc/ccs/compression.tex]: What are these random nonces? Do you want to say just that strings n_1, n_2, n_3 exist? Also this structure mimics reflection too much, but we want our compression methods to be more generic.

File: etc/ccs/theory.tex [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+\begin{document}
+
+\maketitle
+
+\section*{Security under adaptive reflection}
+
+\subsection*{Adaptive reflection game}
+
+Let $\mathcal{PE} = (\mathcal{K}, \mathcal{E}, \mathcal{D})$ be a public-key
+encryption scheme, $\mathcal{A}$ be an adversary and $\mathcal{A'}$ be a
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Use `$\mathcal{S}$` instead of `$\mathcal{A'}$` for the simulator.

File: etc/ccs/theory.tex [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+    while playing:
+        $(st, r_i) \leftarrow \mathcal{A}(\text{"reflect"}, st)$
+        $v_i \leftarrow V$
+        $m_i = f(s, r_i, v_i)$
+        $c_i = \mathcal{E}_{pk}(m_i)$
+        $(st, s^*) \leftarrow \mathcal{A}(\text{"play"}, st, c_i)$
+        if $s^* \neq \bot$:
+            playing = False
+        i += 1
+    if $s^* = s$:
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Use a semantic security property

File: etc/ccs/theory.tex [hunk 13]
```diff
@@ -0,0 +1,225 @@
...
+rounds are needed, and if so, more rounds are played at the discretion of the
+adversary.
+
+When the adversary decides to complete the game, they output a guess $s^*$. The
+adversary is successful if $Q(s) = s^*$.
+
+Formally, let the adversarial game be defined as follows:
+
+\begin{lstlisting}[texcl,mathescape]
+def $\text{Game}_{\text{REF-SEC}}^{\mathcal{PE},\mathcal{A}}(\lambda, f, \mathcal{V}, Q)$:
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Let the adversary choose $Q$

Comment (dionyziz) [inline etc/ccs/theory.tex]: Function $g$

File: etc/ccs/theory.tex [hunk 14]
```diff
@@ -0,0 +1,225 @@
...
+    $(\mathcal{M}, st) \leftarrow \mathcal{A}(\text{"init"}, pk, f, \mathcal{V}, Q)$
+    $s \leftarrow \mathcal{M}$
+    playing = True
+    i = 1
+    while playing:
+        $(st, r_i) \leftarrow \mathcal{A}(\text{"reflect"}, st)$
+        $v_i \leftarrow V$
+        $m_i = f(s, r_i, v_i)$
+        $c_i = \mathcal{E}_{pk}(m_i)$
+        $(st, s^*) \leftarrow \mathcal{A}(\text{"play"}, st, c_i)$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: $s^*$ change to $y$

File: etc/ccs/theory.tex [hunk 15]
```diff
@@ -0,0 +1,225 @@
...
+We now shift our interest to algorithms which compose compression and
+encryption.  Therefore, we treat the public-key encryption scheme above as a
+composition of a compression and encryption algorithm:
+
+\begin{equation*}
+    \mathcal{E} = Enc \circ Com
+\end{equation*}
+
+$Com$ denotes a compression function as defined above, while $Enc$ denotes an
+IND-CPA-secure length-preserving encryption algorithm.
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Split IND-CPA security into separate section.

File: etc/ccs/theory.tex [hunk 16]
```diff
@@ -0,0 +1,225 @@
...
+
+To be length-preserving, $Enc$ must satisfy the following property:
+
+\begin{equation*}
+    \exists c \in \mathbb{N}:
+    \forall m1, m2 \in \{0, 1\}^*:
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: m_1

File: etc/ccs/theory.tex [hunk 17]
```diff
@@ -0,0 +1,225 @@
...
+function's performance, where $E$ denotes expectation.
+
+\subsection*{Compression security}
+
+We now shift our interest to algorithms which compose compression and
+encryption.  Therefore, we treat the public-key encryption scheme above as a
+composition of a compression and encryption algorithm:
+
+\begin{equation*}
+    \mathcal{E} = Enc \circ Com
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Specify that $Enc$ is defined on ${0, 1}*$

File: etc/ccs/theory.tex [hunk 18]
```diff
@@ -0,0 +1,225 @@
...
+
+To be length-preserving, $Enc$ must satisfy the following property:
+
+\begin{equation*}
+    \exists c \in \mathbb{N}:
+    \forall m1, m2 \in \{0, 1\}^*:
+    |m2| - |m1| > c
+    \Rightarrow
+    Enc(m2) > Enc(m1)
+\end{equation*}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Just say that $Enc$ is monotonic

File: etc/ccs/theory.tex [hunk 19]
```diff
@@ -0,0 +1,225 @@
...
+
+\begin{lstlisting}[texcl,mathescape]
+def $\text{Attack}(\lambda, Q, \mathcal{M}, \mathcal{V}, f)$:
+    $(pk, sk) \leftarrow \mathcal{K}(\lambda)$
+    $s \leftarrow \mathcal{M}$
+    $r \leftarrow \mathcal{O}_R(Q)$
+    $v \leftarrow \mathcal{V}$
+    $m = f(s, r, v)$
+    $c = \mathcal{E}_{pk}(m)$
+    $s^* \leftarrow \mathcal{O}_D(c)$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Parameterize with $Q$

File: etc/ccs/theory.tex [hunk 20]
```diff
@@ -0,0 +1,225 @@
...
+
+When the adversary decides to complete the game, they output a guess $s^*$. The
+adversary is successful if $Q(s) = s^*$.
+
+Formally, let the adversarial game be defined as follows:
+
+\begin{lstlisting}[texcl,mathescape]
+def $\text{Game}_{\text{REF-SEC}}^{\mathcal{PE},\mathcal{A}}(\lambda, f, \mathcal{V}, Q)$:
+    $(pk, sk) \leftarrow \mathcal{K}(\lambda)$
+    $(\mathcal{M}, st) \leftarrow \mathcal{A}(\text{"init"}, pk, f, \mathcal{V}, Q)$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: $\mathcal{M}$ should be a parameter in the Game

File: etc/ccs/theory.tex [hunk 21]
```diff
@@ -0,0 +1,225 @@
...
+
+\begin{equation*}
+    \exists c \in \mathbb{N}:
+    \forall m1, m2 \in \{0, 1\}^*:
+    |m2| - |m1| > c
+    \Rightarrow
+    Enc(m2) > Enc(m1)
+\end{equation*}
+
+\subsection*{Reflective attacks}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Reflection

File: etc/ccs/theory.tex [hunk 22]
```diff
@@ -0,0 +1,313 @@
...
+
+\subsection*{Adaptive reflection game}
+
+Let $\mathcal{PE} = (\mathcal{K}, \mathcal{E}, \mathcal{D})$ be a public-key
+encryption scheme, $\mathcal{A}$ be an adversary and $\mathcal{S}$ be a
+simulator of $\mathcal{A}$.  The game
+$\text{Game}_{\text{REF-SEC}}^{\mathcal{PE},\mathcal{A}}(\lambda,  f,
+\mathcal{V}, g)$ is parameterized with a rendering function $f(\cdot, \cdot,
+\cdot)$, a noise distribution $\mathcal{V}$, the security parameter $\lambda$,
+and some plain-text function $g$, and a distribution of secrets $\mathcal{M}$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: and and

File: etc/ccs/theory.tex [hunk 23]
```diff
@@ -0,0 +1,313 @@
...
+\end{equation*}
+
+\subsection*{Adaptive reflection security}
+
+Given a polynomially computable rendering function $f(\cdot, \cdot, \cdot)$ and
+a  noise distribution $\mathcal{V}$, a public-key encryption scheme
+$\mathcal{PE}$ is \textit{reflection-secure} if:
+
+\begin{equation*}
+\forall g:
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: and $\forall \mathcal{M}$

File: etc/ccs/theory.tex [hunk 24]
```diff
@@ -0,0 +1,313 @@
...
+$\mathcal{O}_D(Q)$.  These two algorithms constitute an attack oracle pair
+$(\mathcal{O}_R(Q), \mathcal{O}_D(Q))$.
+
+Specifically, let the attack be defined as follows:
+
+\begin{lstlisting}[texcl,mathescape]
+def $\text{Attack}(\lambda, Q, \mathcal{M}, \mathcal{V}, f)$:
+    $(pk, sk) \leftarrow \mathcal{K}(\lambda)$
+    $s \leftarrow \mathcal{M}$
+    $r \leftarrow \mathcal{O}_R(g)$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: $Q$

File: etc/ccs/theory.tex [hunk 25]
```diff
@@ -0,0 +1,313 @@
...
+    $v \leftarrow \mathcal{V}$
+    $m = f(s, r, v)$
+    $c = \mathcal{E}_{pk}(m)$
+    $y \leftarrow \mathcal{O}_D(Q, c)$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: $b$

File: etc/ccs/theory.tex [hunk 26]
```diff
@@ -0,0 +1,313 @@
...
+\forall s \forall r \forall v_1, v_2 \in \mathcal{V}: f(s, r, v_1) = f(s, r, v_2)
+\end{equation*}
+
+Moreover, we assume that $f$ outputs a concatenation of various instances of
+$s$ and $r$ with arbitrary repetitions, in addition to concatenating with some
+constant context whose position is independent of the values of $s$ and $r$.
+The number of repetitions of $s$ and $r$ are also independent of the values of
+$s$ and $r$.
+
+Formally, let us denote $\Sigma(s)$ and $\Sigma(r)$ the alphabets of $s$ and
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: You don't need these alphabets

File: etc/ccs/theory.tex [hunk 27]
```diff
@@ -0,0 +1,313 @@
...
+We examine the case where $Q$ asks the question of whether the secret is
+exactly equal to a given constant, $x$, which consists of at least three
+symbols.
+
+Then the attack will be constructed witht he following two oracles.
+
+The reflection oracle symbol reflects the candidate secret:
+
+\begin{lstlisting}[texcl,mathescape]
+def $\mathcal{O}_R(c)$:
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: $c -> Q$

File: etc/ccs/theory.tex [hunk 28]
```diff
@@ -0,0 +1,313 @@
...
+        $m_i = f(s, r_i, v_i)$
+        $c_i = \mathcal{E}_{pk}(m_i)$
+        $(st, y) \leftarrow \mathcal{A}(\text{"play"}, st, c_i)$
+        if $y \neq \bot$:
+            playing = False
+        i += 1
+    if $y = g(s)$:
+        return 1
+    return 0
+\end{lstlisting}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Describe symmetric version of the game

File: etc/ccs/theory.tex [hunk 29]
```diff
@@ -0,0 +1,313 @@
...
+Where $\alpha$ is a constant we compare against and is defined as follows:
+
+\begin{equation*}
+\alpha = |Enc_\epsilon(Com(f(X, X, \epsilon)))|
+\end{equation*}
+
+We can now prove that this attack has a non-negligible advantage against a
+simulator.
+
+\subsection*{Compression is insecure}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Number sections

File: etc/ccs/theory.tex [hunk 30]
```diff
@@ -0,0 +1,313 @@
...
+The BREACH attack is a specific instance of reflection-based attacks in which
+the $Q$ property pertains to a prefix of the secret. It asks "Does the secret
+begin with such text?" The reflection algorithm $\mathcal{O}_R(g)$ for BREACH
+then produces the exact prefix being guessed as a reflection string.
+
+We examine the case where $Q$ asks the question of whether the secret is
+exactly equal to a given constant, $x$, which consists of at least three
+symbols.
+
+Then the attack will be constructed witht he following two oracles.
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: spell

File: etc/ccs/theory.tex [hunk 31]
```diff
@@ -0,0 +1,313 @@
...
+\end{lstlisting}
+
+Where $\alpha$ is a constant we compare against and is defined as follows:
+
+\begin{equation*}
+\alpha = |Enc_\epsilon(Com(f(X, X, \epsilon)))|
+\end{equation*}
+
+We can now prove that this attack has a non-negligible advantage against a
+simulator.
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Proof

File: etc/ccs/theory.tex [hunk 32]
```diff
@@ -0,0 +1,313 @@
...
+Rep(x:xs, \alpha, \beta) &\defeq x:Rep(xs, \alpha, \beta)
+\end{align*}
+
+Where $:$ denotes the concatenation of a symbol with a string.
+
+We will first concern ourselves with the following specific instance of $f$:
+
+\begin{equation*}
+\begin{split}
+f(s, r, v) = s || . || v\\
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: $s || . || r$

File: etc/ccs/theory.tex [hunk 33]
```diff
@@ -0,0 +1,313 @@
...
+\end{equation*}
+
+Our assumptions for $f$ can be stated as follows:
+
+\begin{equation*}
+\begin{split}
+\exists \mu \in \Sigma\diamond^*\\
+\exists u, v, w, z \in \Sigma\diamond^*\\
+\mu = u || \diamond || v \land
+\mu = w || \star || v \land\\
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: v -> z

File: etc/ccs/theory.tex [hunk 34]
```diff
@@ -0,0 +1,313 @@
...
+\begin{lstlisting}[texcl,mathescape]
+def $\mathcal{O}_R(c)$:
+    return $X$
+\end{lstlisting}
+
+The distinguisher oracle compares lengths:
+
+\begin{lstlisting}[texcl,mathescape]
+def $\mathcal{O}_D(c)$:
+    return $|c| > \alpha$
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: <=

File: etc/ccs/theory.tex [hunk 35]
```diff
@@ -0,0 +1,460 @@
...
+
+Let
+
+\begin{equation*}
+    \text{Adv}_{\mathcal{PE}, \mathcal{A}, \mathcal{S}}(\lambda, f, \mathcal{V}, \mathcal{M}, g)
+    \defeq
+    Pr[\text{Game}_{\text{REF-SEC}}^{\mathcal{PE},\mathcal{A}}(\lambda, f, \mathcal{V}, \mathcal{M}, g) = 1]
+    -
+    Pr[\text{Game}_{\text{REF-SIM}}^{\mathcal{PE},\mathcal{S}}(\lambda, f, \mathcal{V}, \mathcal{M}, g) = 1].
+\end{equation*}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Advantage needs ||

File: etc/ccs/theory.tex [hunk 36]
```diff
@@ -0,0 +1,460 @@
...
+\begin{equation*}
+    \begin{split}
+        \exists y \in Y\\
+        Pr[
+            |Com(f(x_1, y, v_1))| < |Com(f(x_2, y, v_2))|
+        ]
+        \geq
+        \frac{1}{2} + \alpha(\lambda)
+    \end{split}
+\end{equation*}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: There is an efficient distinguisher? Use statistical distance.

File: etc/ccs/theory.tex [hunk 37]
```diff
@@ -0,0 +1,460 @@
...
+reflection, and the noise, between some context strings which constitute the
+static portions of the site.
+
+We assume that $f$ outputs a concatenation of various instances of
+$s$, $r$, and $v$ with arbitrary repetitions, in addition to concatenating with some
+constant context whose position is independent of the values of $s, r, v$.
+The number of repetitions of $s, r, v$ are also independent of the values of
+$s, r, v$.
+
+Formally, assume that the symbols $\{\beta, \gamma, \delta\}$ do not appear in the
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: \beta_1, ...

File: etc/ccs/theory.tex [hunk 38]
```diff
@@ -0,0 +1,460 @@
...
+\begin{lemma}
+
+Let $f$ be a concatenative rendering function and $Com = LZ77$. Then
+let $Q$ be a $k$-length prefix checking predicate.
+
+\begin{equation}
+Pr[Q \textrm{ compression-detectable under } f, Com] = \text{non-negl}(\lambda - k)
+\end{equation}
+
+Where the probability randomness is taken over predicates $Q$.
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Parameterize $Q$ with the prefix.

File: etc/ccs/theory.tex [hunk 39]
```diff
@@ -0,0 +1,460 @@
...
+\end{equation}
+
+But the probability of success of $\mathcal{A}$ is $\frac{1}{2} + \alpha(\lambda)$ for some non-negligible function $\alpha$ (proof?).
+
+Therefore,
+
+\begin{equation}
+\begin{split}
+    \forall PPT \mathcal{S}:
+    \text{Adv}_{\mathcal{SE}(Enc, LZ77), \mathcal{A}, \mathcal{S}}(\lambda, f, \mathcal{V}, \mathcal{M}, g) \geq ?
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Com

**Comment by dionyziz** [comment]
Full recovery attack

**Comment by dionyziz** [comment]
Describe defences in the wild

**Comment by dionyziz** [comment]
Positive result?

File: etc/ccs/theory.tex [hunk 40]
```diff
@@ -0,0 +1,460 @@
...
+    $l_1 = |Enc_\epsilon(Com(f(x_1, y, v_1)))|$
+    $l_2 = |Enc_\epsilon(Com(f(x_2, y, v_2)))|$
+    
+    if $|c| \leq l_1$:
+        return True
+    if $|c| \geq l_2$:
+        return False
+        
+    return $\bot$
+\end{lstlisting}
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Use BPP amplification to boost probability of success

File: etc/ccs/theory.tex [hunk 41]
```diff
@@ -0,0 +1,460 @@
...
+
+    $v_1 \leftarrow \mathcal{V}$
+    $v_2 \leftarrow \mathcal{V}$
+
+    $l_1 = |Enc_\epsilon(Com(f(x_1, y, v_1)))|$
+    $l_2 = |Enc_\epsilon(Com(f(x_2, y, v_2)))|$
+    
+    if $|c| \leq l_1$:
+        return True
+    if $|c| \geq l_2$:
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: else

File: etc/ccs/theory.tex [hunk 42]
```diff
@@ -0,0 +1,460 @@
...
+
+\section{The security of compression}
+
+\subsection{Compression functions}
+
+A compression function $Com$ is a polynomial polynomially reversible $1 - 1$
+function that takes an input $x$ and outputs $y$. $x$ is a random variable that
+follows the distribution $X$. $y = Com(x)$ will follow the
+distribution $Com(X)$.  $E(|y|) / E(|x|) < 1$ is the metric that defines the
+function's performance, where $E$ denotes expectation.
```
Comment (dionyziz) [inline etc/ccs/theory.tex]: Not sure we still need this definition

File: etc/ccs/theory.tex [hunk 43]
```diff
@@ -0,0 +1,460 @@
...
+
+def $\mathcal{A}("play", Q, \mathcal{M}, c, y)$:
+    $x_1 \leftarrow \mathcal{M}|_Q$
+    $x_2 \leftarrow \mathcal{M}|_{\lnot Q}$
+
+    $v_1 \leftarrow \mathcal{V}$
+    $v_2 \leftarrow \mathcal{V}$
+
+    $l_1 = |Enc_\epsilon(Com(f(x_1, y, v_1)))|$
+    $l_2 = |Enc_\epsilon(Com(f(x_2, y, v_2)))|$
```
Comment (dimkarakostas) [inline etc/ccs/theory.tex]: replace x, y with s, r for consistency with 1.1

**Pull request merged**